File: SkyEditor.Core.Windows/ConsoleCommands/DistPrep.vb
Imports SkyEditor.Core.ConsoleCommands
Imports SkyEditor.Core.Utilities

Namespace ConsoleCommands
    Public Class DistPrep
        Inherits ConsoleCommand

        Protected Overrides Sub Main(Arguments() As String)
            RedistributionHelpers.PrepareForDistribution(CurrentApplicationViewModel.CurrentPluginManager)
        End Sub
    End Class
End Namespace

File: SkyEditor.Core/ConsoleCommands/DistPrep.cs
using SkyEditor.Core.IO;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SkyEditor.Core.ConsoleCommands
{
    public class DistPrep : ConsoleCommand
    {
        public DistPrep(PluginManager pluginManager, IIOProvider provider) : base(provider)
        {
            CurrentPluginManager = pluginManager;
        }
        
        protected PluginManager CurrentPluginManager { get; }

        public override async Task MainAsync(string[] arguments)
        {
            CurrentPluginManager.PreparePluginsForDistribution();
            await base.MainAsync(arguments);
        }
    }
}
